src/cloudWatchLogs/cloudWatchLogsUtils.ts

```diff
@@ -2,17 +2,38 @@
  * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
  * SPDX-License-Identifier: Apache-2.0
  */
-
+import * as telemetry from '../shared/telemetry/telemetry'
 import * as vscode from 'vscode'
 import { CLOUDWATCH_LOGS_SCHEME } from '../shared/constants'
 import { fromExtensionManifest } from '../shared/settings'
-import { CloudWatchLogsGroupInfo } from './registry/logStreamRegistry'
+import { CloudWatchLogsData, CloudWatchLogsGroupInfo } from './registry/logStreamRegistry'
 import { CloudWatchLogsParameters } from './registry/logStreamRegistry'
 
 // URIs are the only vehicle for delivering information to a TextDocumentContentProvider.
 // The following functions are used to structure and destructure relevant information to/from a URI.
 // Colons are not valid characters in either the group name or stream name and will be used as separators.
 
+/**
+ * records a metric that the filter was successful IF a filter was actually applied i.e one of time or filterPattern were set.
+ * @param logData
+ * @param resourceType
+ */
+export function telemetryFilter(
+    logData: CloudWatchLogsData,
+    resourceType: telemetry.CloudWatchResourceType,
+    source: 'escapeHatch' | 'originalSearch'
```

Comment:
These two terms are kind of specific to the implementation. Can we come up with names closer to the UI?
Isn't `escapeHatch` the same as 'Editor' (or 'EditorButton')? And isn't `originalSearch` from the command palette?

Reply:
I agree that sounds like a good idea. The `originalSearch` isn't always from the palette, it could be from the explorer or anywhere the search is accessible. I don't believe I have the source of the search at this point, but I didn't think it was incredibly relevant. Should I distinguish this or label it `Search` or `OriginalSearch`?

Reply:
I guess you can put "Command" as I consider that to be the most generic source.